[Requirements] Bump the version of Pytest and other testing dependencies

jond01 opened the pull request:
Resolves [ML-6658](https://iguazio.atlassian.net/browse/ML-6658).
Pytest 8 is available, we use 7.

[ML-6658]: https://iguazio.atlassian.net/browse/ML-6658?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jond01 committed:
[Requirements] Bump the version of Pytest and other testing dependencies

Pytest 8 is available, we use 7.

jond01 commented on dev-requirements.txt:
```diff
@@ -7,17 +7,17 @@ ruff==0.4.2
 pytest-timeout~=2.0
 aioresponses~=0.7
 requests-mock~=1.8
 httpx~=0.24.0
```
I could not bump the version of `httpx` to `~=0.27.0` due to `igz-mgmt`:
```
ERROR: Cannot install -r dockerfiles/mlrun-api/requirements.txt (line 7), httpx~=0.27 and igz-mgmt==0.1.3 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested httpx~=0.27
    igz-mgmt 0.1.3 depends on httpx~=0.24.1
    The user requested httpx~=0.27
    igz-mgmt 0.1.4 depends on httpx~=0.24.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

liranbg replied:
@moranbental  fyi

moranbental replied:
I will bump the httpx version in the igz-mgmt

moranbental replied:
This requires dropping support for Python 3.7 in the igz-mgmt

jond01 replied:
I think that if you loosen the specification to allow every `0.x.y` version (`httpx~=0.24` in igz-mgmt) it will support both Python 3.7 and newer httpx versions in mlrun

moranbental replied:
please change the `igz-mgmt` version to `0.2.0`

jond01 replied:
Thanks, will do.

jond01 committed:
Preserve pytest-asyncio auto mode

See:
https://pytest-asyncio.readthedocs.io/en/latest/concepts.html#auto-mode
https://pytest-asyncio.readthedocs.io/en/latest/reference/changelog.html#id46

jond01 committed:
Remove async from fixtures that don't need it

Fix the decorators and type hints of the true
async fixtures.

jond01 committed:
Explicitly set pytest-asyncio strict mode

jond01 committed:
Downgrade pytest-asyncio to 0.22

There's some issue with 0.23, see:
https://pytest-asyncio.readthedocs.io/en/latest/reference/changelog.html#known-issues

jond01 committed:
Use absolute path

jond01 committed:
Use tmpdir fixture instead of TemporaryDirectory

jond01 committed:
Avoid in-memory store cross-contamination

jond01 committed:
Change asyncio_mode="auto" to avoid confusion
